src/pages/index.js

```diff
@@ -0,0 +1,374 @@
+  api
+    .editUserInfo({
+      name: editProfileNameInput.value,
+      about: editProfileDescriptionInput.value,
+    })
+    .then((data) => {
+      profileNameEl.textContent = data.name;
+      profileDescriptionEl.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      editProfileSubmitBtn.textContent = "Save";
+    });
+}
+
+// NEW POST FUNCTION
+
+function handleNewPostSubmit(evt) {
+  evt.preventDefault();
+  const newPostSubmitBtn = addCardForm.querySelector(".modal__submit-btn");
+  newPostSubmitBtn.textContent = "Saving...";
+
+  const inputValues = {
+    name: newPostCaption.value,
+    link: newPostLink.value,
+  };
+
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(newPostSubmitBtn, settings);
+      closeModal(newPostModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      newPostSubmitBtn.textContent = "Create";
+    });
+}
+
+// DELETE CARD FUNCTIONS
+
+let selectedCard = null;
+let selectedCardId = null;
+
+function openConfirmDeleteModal(cardElement) {
+  selectedCard = cardElement;
+  selectedCardId = cardElement.dataset.cardId;
+  openModal(confirmDeleteModal);
+}
+
+confirmDeleteBtn.addEventListener("click", () => {
+  confirmDeleteBtn.textContent = "Deleting...";
+  if (selectedCard && selectedCardId) {
+    api
+      .deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        selectedCard = null;
+        selectedCardId = null;
+        closeModal(confirmDeleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        confirmDeleteBtn.textContent = "Delete";
+      });
+  }
+});
+
+cancelDeleteBtn.addEventListener("click", () => {
+  selectedCard = null;
+  selectedCardId = null;
+  closeModal(confirmDeleteModal);
+});
+
+confirmDeleteCloseBtn.addEventListener("click", () =>
+  closeModal(confirmDeleteModal)
+);
+
+// EVENT LISTENERS
+
+// Edit profile modal
+editProfileBtn.addEventListener("click", () => {
+  editProfileNameInput.value = profileNameEl.textContent;
+  editProfileDescriptionInput.value = profileDescriptionEl.textContent;
+  resetValidation(
+    editProfileForm,
+    [editProfileNameInput, editProfileDescriptionInput],
+    settings
+  );
+  openModal(editProfileModal);
+});
+
+editProfileCloseBtn.addEventListener("click", () =>
+  closeModal(editProfileModal)
+);
+editProfileForm.addEventListener("submit", handleEditProfileSubmit);
+
+// New post modal
+newPostBtn.addEventListener("click", () => openModal(newPostModal));
+newPostCloseBtn.addEventListener("click", () => closeModal(newPostModal));
+addCardForm.addEventListener("submit", handleNewPostSubmit);
+
+// Edit avatar modal
+avatarModalBtn.addEventListener("click", () => {
+  avatarForm.reset();
+  resetValidation(avatarForm, [avatarLink], settings);
```

Comment:
Please, don't reset the `avatar` form if you don't insert the link into the input, otherwise the button is **active**, but the input is empty https://snipboard.io/0pfsad.jpg